leshan-server-core/src/main/java/org/eclipse/leshan/server/security/oscore/OscoreSetting.java

```diff
@@ -18,27 +18,43 @@
 import java.io.Serializable;
 import java.util.Arrays;
 
+import org.eclipse.leshan.core.oscore.AeadAlgorithm;
+import org.eclipse.leshan.core.oscore.HkdfAlgorithm;
 import org.eclipse.leshan.core.oscore.OscoreIdentity;
 import org.eclipse.leshan.core.util.Hex;
+import org.eclipse.leshan.core.util.Validate;
 
+/**
+ * OSCORE Settings.
+ * <p>
+ * See : https://datatracker.ietf.org/doc/html/rfc8613#section-3.2
+ */
 public class OscoreSetting implements Serializable {
     private static final long serialVersionUID = 1L;
 
     private final byte[] senderId;
     private final byte[] recipientId;
     private final byte[] masterSecret;
-    private final Integer aeadAlgorithm;
-    private final Integer hmacAlgorithm;
+    private final AeadAlgorithm aeadAlgorithm;
+    private final HkdfAlgorithm hkdfAlgorithm;
     private final byte[] masterSalt;
 
-    public OscoreSetting(byte[] senderId, byte[] recipientId, byte[] masterSecret, Integer aeadAlgorithm,
-            Integer hmacAlgorithm, byte[] masterSalt) {
+    public OscoreSetting(byte[] senderId, byte[] recipientId, byte[] masterSecret) {
+        this(senderId, recipientId, masterSecret, null, null, null);
+    }
+
+    public OscoreSetting(byte[] senderId, byte[] recipientId, byte[] masterSecret, AeadAlgorithm aeadAlgorithm,
+            HkdfAlgorithm hkdfAlgorithm, byte[] masterSalt) {
+        Validate.notNull(senderId);
+        Validate.notNull(recipientId);
+        Validate.notNull(masterSecret);
+        // TODO OSCORE add maximum length check
         this.senderId = senderId;
         this.recipientId = recipientId;
         this.masterSecret = masterSecret;
-        this.aeadAlgorithm = aeadAlgorithm;
-        this.hmacAlgorithm = hmacAlgorithm;
-        this.masterSalt = masterSalt;
+        this.aeadAlgorithm = aeadAlgorithm == null ? AeadAlgorithm.AES_CCM_16_64_128 : aeadAlgorithm;
+        this.hkdfAlgorithm = hkdfAlgorithm == null ? HkdfAlgorithm.HKDF_HMAC_SHA_256 : hkdfAlgorithm;
+        this.masterSalt = masterSalt == null ? new byte[0] : masterSalt;
```

Comment:
@rikard-sics 

Is there some validation we can do here ? 

I see at https://datatracker.ietf.org/doc/html/rfc8613#section-3.1
> Sender ID.  Byte string used to identify the Sender Context, to
      derive AEAD keys and Common IV, and to contribute to the
      uniqueness of AEAD nonces.  Maximum length is determined by the
      AEAD Algorithm.
      
and 

> Recipient ID.  Byte string used to identify the Recipient Context,
      to derive AEAD keys and Common IV, and to contribute to the
      uniqueness of AEAD nonces.  Maximum length is determined by the
      AEAD Algorithm.
      
I don't know which data can be used from https://datatracker.ietf.org/doc/html/rfc8152#section-10 to define max length.

Is there limitation for `masterSalt` and/or `masterKey`, I don't think so but I prefer to be sure :thinking:

Reply:
In the OSCORE code in Californium there are calculations related to this. See the following:
https://github.com/eclipse/californium/blob/0459aa8c21eb59137cb6460fe352d0b73a1173d4/cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java#L686-L688
https://github.com/eclipse/californium/blob/0459aa8c21eb59137cb6460fe352d0b73a1173d4/cf-oscore/src/main/java/org/eclipse/californium/cose/EncryptCommon.java#L217

See also the following sections in the OSCORE RFC:
https://datatracker.ietf.org/doc/html/rfc8613#section-5.2
https://datatracker.ietf.org/doc/html/rfc8613#section-3.3

> The maximum length of Sender ID in
> bytes equals the length of the AEAD nonce minus 6
> For AES-CCM-16-64-128 the maximum length of Sender ID is 7 bytes.

As for the master secret and master salt there is no maximum length enforced by OSCORE.

Reply:
Great ! thx for the answers :pray: 
I added this kind of validation in the last commit 08690ee